src/generator/index-collection-helpers.ts

```diff
@@ -0,0 +1,42 @@
+import { IndexCollection } from './types';
```

Comment:
I personally prefer working with a plain object rather than an array - especially for this implementation, where you iterate the array constantly to identify an entry by a property that could be the key in a plain object